src/python_interpreter.cc

```diff
@@ -0,0 +1,81 @@
+#include "python_interpreter.h"
+#include "notebook.h"
+#include "config.h"
+#include <iostream>
+#include <pygobject.h>
+
+inline pybind11::module pyobject_from_gobj(gpointer ptr){
+  auto obj=G_OBJECT(ptr);
+  if(obj)
+    return pybind11::module(pygobject_new(obj), false);
+  return pybind11::module(Py_None, false);
+}
+
+PythonInterpreter::PythonInterpreter(){
+  auto init_juci_api=[](){
+    pybind11::module(pygobject_init(-1,-1,-1), false);
+    pybind11::module api("jucpp", "Python bindings for juCi++");
+    api.def("get_current_text_buffer", [](){
+      auto view=Notebook::get().get_current_view();
+      if(view)
+        return pyobject_from_gobj(view->gobj());
+      return pybind11::module(Py_None, false);
+    });
+    return api.ptr();
+  };
+  PyImport_AppendInittab("jucipp", init_juci_api);
+  Py_Initialize();
+  Config::get().load();
+}
+
+pybind11::module PythonInterpreter::get_loaded_module(const std::string &module_name){
+  return pybind11::module(PyImport_AddModule(module_name.c_str()), true);
+}
+
+pybind11::module PythonInterpreter::import(const std::string &module_name){
+  return pybind11::module(PyImport_ImportModule(module_name.c_str()), false);
+}
+
+void PythonInterpreter::add_path(const boost::filesystem::path &path){
+  std::wstring sys_path(Py_GetPath());
+  if(!sys_path.empty())
+#ifdef _WIN32
+    sys_path += ';';
+#else
+    sys_path += ':';
+#endif
+  sys_path += path.generic_wstring();
+  Py_SetPath(sys_path.c_str());
+}
+
+PythonInterpreter::~PythonInterpreter(){
+  auto err=PythonError();
+  if(Py_IsInitialized())
+    Py_Finalize();
+  if(err)
+    std::cerr << std::string(err) << std::endl;
+}
+
+PythonError::PythonError(){
```

Comment:
@zalox Have not studied the changes in detail, just saw that you repeated Python in for instance PythonInterpreter and PythonError. I think its better to use namespaces such that it becomes: Python::Interpreter and Python::Error.